h2o-algos/src/main/java/hex/tree/SharedTree.java

```diff
@@ -409,10 +409,10 @@ protected final void resumeFromCheckpoint(SharedTree st) {
     protected final void scoreAndBuildTrees(boolean oob) {
       for( int tid=0; tid< _ntrees; tid++) {
         // During first iteration model contains 0 trees, then 1-tree, ...
-        boolean scored = doScoringAndSaveModel(false, oob, _parms._build_tree_one_node);
+        final boolean scored = doScoringAndSaveModel(false, oob, _parms._build_tree_one_node);
         if (scored && ScoreKeeper.stopEarly(_model._output.scoreKeepers(), _parms._stopping_rounds, getProblemType(), _parms._stopping_metric, _parms._stopping_tolerance, "model's last", true)) {
-          doScoringAndSaveModel(true, oob, _parms._build_tree_one_node);
           _job.update(_ntrees-_model._output._ntrees); //finish
+          LOG.info(_model.toString()); // we don't know if doScoringAndSaveModel printed the model or not
```

Comment:
I am just curious, why we don't know if doScoringAndSaveModel printed the model or not?

Reply:
Because not every scoring iteration prints out the model, this is to avoid too much output in the log.

We don't know if this condition was true or not:

      final boolean printout = (_parms._score_each_iteration || finalScoring || sinceLastScore > _parms._score_interval);

It is fine though, worst case we will print the model twice as before (which doesn't change with this PR).

Reply:
Ok, thanks for the explanation.